src/Microsoft.SqlTools.ServiceLayer/DacFx/GenerateDeployPlanOperation.cs

```diff
@@ -0,0 +1,125 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlServer.Dac;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.DacFx.Contracts;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using Microsoft.SqlTools.Utility;
+using System;
+using System.Data.SqlClient;
+using System.Diagnostics;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx
+{
+    /// <summary>
+    /// Class to represent an in-progress generate deploy plan operation
+    /// </summary>
+    class GenerateDeployPlanOperation : DacFxOperation
+    {
+        public GenerateDeployPlanParams Parameters { get; }
+
+        public GenerateDeployPlanOperation(GenerateDeployPlanParams parameters, ConnectionInfo connInfo): base(connInfo)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+        }
+
+        public override void Execute()
+        {
+        }
+
+        public string ExecuteGenerateDeployReport()
+        {
+            DacServices ds = new DacServices(this.ConnectionString);
+            DacPackage dacpac = DacPackage.Load(this.Parameters.PackageFilePath);
+            DacDeployOptions options = GetDefaultDeployOptions();
+            // this is so that operations with possible data loss will be in the report
+            options.BlockOnPossibleDataLoss = false;
+
+            return ds.GenerateDeployReport(dacpac, this.Parameters.DatabaseName, options, this.CancellationToken);
+        }
+
+        private DacDeployOptions GetDefaultDeployOptions()
+        {
+            DacDeployOptions options = new DacDeployOptions
```

Comment:
I'd had to have defaults hard coded since they can change. Do we have to specify them here? Are the default not already set when instantiating a new instance?

Reply:
These are the defaults used in SSMS, which are different from the default DacDeployOptions when instantiating a new instance. For example, the default for DropObjectsNotInSource is false, but in SSMS the default is true.

Reply:
That's unfortunate. sqlpackage, DacFx API, and now SSMS have different defaults. Are you proposing SSMS should be defaults going forward?

Reply:
It would be good to have the same defaults for all three, but I don't know if the SSMS values should be the default since it could result in unintentionally dropping objects. In this case, using the SSMS values makes sense so that the same deploy report is generated in both SSMS and ADS.

Reply:
Have any proposals to resolve this?

Reply:
Most of the SSMS default options are the same as the DacFx ones, so here this can be changed to only set the options that are different rather than setting all of them. It looks like only 7 of them are different from the default value set. It would be good to also change this in SSMS to simplify.

Reply:
This works for now. Let's make sure that we use the same options when running the deployment in the deploy operation. Long term, I think we should align on the DacFx defaults, and make that change in SSMS. This was, we specify the defaults in one place.